### PR TITLE
Feature/content extract improvement

### PR DESCRIPTION
**Pull request by tomaszmichalak**
---

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: knotx-core/pom.xml [hunk 1]
```diff
@@ -100,6 +100,15 @@
       <groupId>uk.co.datumedge</groupId>
       <artifactId>hamcrest-json</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-all</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-all</artifactId>
```
Comment (wiiitek) [inline knotx-core/pom.xml]: Mockito has `provided` scope in *knotx-junit* module.
It looks like in core module the default scope could be used (test).
Or am I wrong? Could we remove thes scope override here?
(should have)

File: knotx-core/src/test/java/io/knotx/fragments/FragmentContentExtractorTest.java [hunk 2]
```diff
@@ -37,8 +42,22 @@
   public void getUnwrappedContent_withFragment_expectDefinedContent(Fragment fragment,
```
Comment (toniedzwiedz) [inline knotx-core/src/test/java/io/knotx/fragments/FragmentContentExtractorTest.java]: Looking at the test input, it seems we're only testing the happy paths here. How is the class under test supposed to behave on invalid input?

File: knotx-knot/knotx-knot-fragment-assembler/src/test/java/io/knotx/assembler/FragmentAssemblerTest.java [hunk 3]
```diff
@@ -67,7 +72,7 @@ public void callAssemblerWithNoSnippets_expectInternalServerError(TestContext co
   @KnotxConfiguration("test.unwrap.io.knotx.FragmentAssembler.json")
   public void callAssemblerWithEmptySnippet_expectNoContentStatus(TestContext context)
       throws Exception {
-    callAssemblerWithAssertions(context, Collections.singletonList(" "),
+    callAssemblerWithAssertions(context, Collections.singletonList(new ImmutablePair<>(Collections.singletonList(RAW), " ")),
```
Comment (toniedzwiedz) [inline knotx-knot/knotx-knot-fragment-assembler/src/test/java/io/knotx/assembler/FragmentAssemblerTest.java]: What is the role of the space being passed as a string here?

File: knotx-core/src/test/java/io/knotx/fragments/FragmentContentExtractorTest.java [hunk 4]
```diff
@@ -37,8 +42,22 @@
...
       String expectedContentFileName) throws Exception {
 
-    final String expectedContent = FileReader.readText(expectedContentFileName);
+    final String expectedContent = readText(expectedContentFileName);
     final String unwrappedContent = FragmentContentExtractor.getUnwrappedContent(fragment);
     assertThat(unwrappedContent, equalToIgnoringWhiteSpace(expectedContent));
   }
+
+  @Coercion
+  public Fragment provideFragment(String fragmentContentFile) throws IOException {
```
Comment (tomaszmichalak) [inline knotx-core/src/test/java/io/knotx/fragments/FragmentContentExtractorTest.java]: The problem here is connected with circular dependency between knotx-core and  knotx-junit. Can we fix this issue in new PR?

File: knotx-knot/knotx-knot-fragment-assembler/src/test/java/io/knotx/assembler/FragmentAssemblerTest.java (hunk 3 above)
Comment (tomaszmichalak) [inline knotx-knot/knotx-knot-fragment-assembler/src/test/java/io/knotx/assembler/FragmentAssemblerTest.java]: Blank string. Fixed to StringUtils.SPACE.

File: knotx-core/pom.xml (hunk 1 above)
Comment (tomaszmichalak) [inline knotx-core/pom.xml]: Fixed.

File: knotx-knot/knotx-knot-fragment-assembler/src/test/java/io/knotx/assembler/FragmentAssemblerTest.java (hunk 3 above)
Comment (toniedzwiedz) [inline knotx-knot/knotx-knot-fragment-assembler/src/test/java/io/knotx/assembler/FragmentAssemblerTest.java]: @tomaszmichalak TBH, `StringUtils.SPACE` is not at all better than `" "`

It's just as meaningless in this context. I'd rather introduce a constant describing its purpose. Is it a separator?

**Pull request merged**